File: app/src/main/java/net/sunniwell/app/linktaro/nettv/content/PageBase.java
package net.sunniwell.app.linktaro.nettv.content;

import android.content.Context;
import android.os.Message;
import android.widget.LinearLayout;

public abstract class PageBase extends LinearLayout {
    public abstract void doByMessage(Message message, int i);

    public abstract void onCreate(Context context);

    public abstract void onDestroy();

    public abstract void onResume();

    public abstract void onStop();

    public PageBase(Context context) {
        super(context);
        onCreate(context);
    }
}

File: app/src/main/java/com/eliving/tv/ui/launcher/nettv/content/PageBase.kt
package com.eliving.tv.ui.launcher.nettv.content

import android.content.Context
import android.os.Message
import android.widget.LinearLayout

abstract class PageBase(context: Context?) :
    LinearLayout(context) {
    abstract fun doByMessage(message: Message?, i: Int)
    abstract fun onCreate(context: Context?)
    abstract fun onDestroy()
    abstract fun onResume()
    abstract fun onStop()

    init {
        onCreate(context)
    }
}